### ISSUE TITLE
Invalid line number in runner generator in very specific case

### ISSUE THREAD

**Issue by trond-snekvik** [ThrowTheSwitch/Unity#288]
If I create two test functions where the name of the second is a subset of the first, the second will get reported with the same line as the first in the test runner. To reproduce, make a test with these two test functions:
```C
void test_my_function_invalid_behavior(void)
{}

void test_my_function(void)
{}
```
Creates output
```
test_file.c:test_my_function_invalid_behavior:11:PASS
test_file.c:test_my_function:11:PASS
```

Note that the second must be a subset of the first, and there can't be any other test-functions between them.

The issue comes from this line:

https://github.com/ThrowTheSwitch/Unity/blob/a868b2eb73ff36ef219383e5f93a30f82ec57eaa/auto/generate_test_runner.rb#L122

This regex either needs an added white space and opening parenthesis here, to ensure that we're matching to the end of the function, or the source index iteration should be `source_index += index + 1` on line 123, to avoid matching on the same function twice.

**Comment by mvandervoord** [comment]
Thanks for the thorough investigation! I'll get that rolled into the next release.

**Comment by mvandervoord** [comment]
Fixed in version in f278c18
